core/shared/src/main/scala/zio/prelude/ZValidation.scala

```diff
@@ -150,6 +170,12 @@ sealed trait ZValidation[+W, +E, +A] { self =>
   final def toEither[E1 >: E]: Either[NonEmptyMultiSet[E1], A] =
     fold(Left(_), Right(_))
 
+  /**
+   * Transforms this `ZValidation` to an `Either`, transforming the accumulated errors and discarding the log.
+   */
+  final def toEitherWith[E1 >: E, E2](f: NonEmptyMultiSet[E1] => E2): Either[E2, A] =
```

Comment:
Btw, is `E1 >: E` the correct thing to do here? I'm never sure with this :laughing:

Reply:
No, if `E` just appears as an input into a function like here you can use `E` and it can avoid type inference issues in some cases.

Reply:
Fixed. Thanks!